feat(ActionSheet): new prop to allow keyboard to stay open on press when desired

breynolds-dev opened the pull request:
Ran into a small issue using an input inside of the action sheet with something like a save button - after typing in the input the keyboard would disappear but not let my press carry through to the button - instead it simply closed the keyboard.  

To the best of my knowledge this is due to the ScrollView and the way it handles taps normally, but adding `keyboardWillPersistTaps` to the ScrollView allowed it to operate as expected for the user.

Instead of forcing it on all the time I created a new prop that would accept the same options `keyboardWillPersistTaps` does and allow the consumer to decide if they want it or not.  Should allow for maximum backwards compatibility and the flexibility to control the keyboard in any way I desire as a consumer.

I was a little unsure of the best prop name to use here, this one is a little verbose, but hopefully clear.  I can change or update this as desired

|  Without keyboardWillPersistTaps | With keyboardWillPersistTaps  |
|---|---|
|  ![2020-05-21 16 25 58](https://user-images.githubusercontent.com/13542564/82608238-e72f2580-9b7f-11ea-8dc0-3c222a4e07a2.gif) | ![2020-05-21 16 26 19](https://user-images.githubusercontent.com/13542564/82608361-1e053b80-9b80-11ea-9d1a-845ba97ba033.gif)  |

breynolds-dev commented:
While this worked fine when I hard coded it, I'm having trouble with the prop passing the value as expected, not sure what's happening but I'm investigating - sorry for the mess up - first MR and all that for me to another repo :)

breynolds-dev commented:
I figured out what I was doing wrong and it was rather dumb on my part.  Turns out it does work as expected when you pass the props to the right component

ammarahm-ed commented:
What if we called it `keyboardWillPersistTaps`. I think that would make more sense instead of a new name? I hope you can change it to that.

ammarahm-ed commented:
@breynolds-dev thanks for updating the prop-name. however I looked at the types file, it has some errors, also the `keyboardWillPersistTaps` is missing. And you have removed the JSDoc for `indicatorColor`. While you fix these, please update README.md file too with the new prop so I can publish a new release once I merge this.

breynolds-dev commented:
Sorry about that I think I was overriding it by accident with my changes.  I updated the types file and the readme with the new prop name, a short description and it's defaults.  Please let me know if you still see anything wrong.  Sorry for all the trouble :)

I also squashed the commits down

ammarahm-ed commented:
@breynolds-dev thank you for fixing this. i will merge it soon! Looks good now

breynolds-dev commented:
Thanks for the review and for the project itself!  It's been really good to work with!

ammarahm-ed commented:
Just pushed the new release v0.2.8 on npm
